File: manual/src/test/java/examples/java/YamlJacksonDocTest.java
package examples.java;

import static examples.ExampleClass.createPerson;
import static org.approvej.ApprovalBuilder.approve;
import static org.approvej.yaml.jackson.YamlPrinter.yamlPrinter;

import examples.ExampleClass.Person;
import java.time.LocalDate;
import org.junit.jupiter.api.Test;

class YamlJacksonDocTest {

  @Test
  void print_yaml() {
    // tag::print_yaml[]
    Person person = createPerson("John Doe", LocalDate.of(1990, 1, 1));

    approve(person)
        .printWith(yamlPrinter()) // <1>
        .byFile();
    // end::print_yaml[]
  }
}

File: manual/src/test/kotlin/examples/kotlin/YamlJacksonDocTest.kt
package examples.kotlin

import examples.ExampleClass.createPerson
import java.time.LocalDate
import org.approvej.ApprovalBuilder.approve
import org.approvej.yaml.jackson.YamlPrinter.yamlPrinter
import org.junit.jupiter.api.Test

class YamlJacksonDocTest {

  @Test
  fun printYaml() {
    // tag::print_yaml[]
    val person = createPerson("John Doe", LocalDate.of(1990, 1, 1))

    approve(person)
      .printedWith(yamlPrinter()) // <1>
      .byFile()
    // end::print_yaml[]
  }
}
